README.md

```diff
@@ -55,6 +55,110 @@ for await (const data of body) {
 console.log('trailers', trailers)
 ```
 
+## `undici.request(url[, options]): Promise`
+
+Arguments:
+
+* **url** `string | URL | object`
+* **options** [`RequestOptions`]
+  * **dispatcher** `Dispatcher` - Default: [getGlobalDispatcher]
+  * **method** `String` - Default: `GET`
+* **maxRedirections** `Integer` - Default: `0`
+
+Returns a promise with the result of the `Dispatcher.request` method.
+
+`url` may contain pathname. `options` may not contain path.
+
+Calls `options.dispatcher.request(options)`.
+
+See [Dispatcher.request] for more details.
+
+## `undici.stream(url, options, factory): Promise`
+
+Arguments:
+
+* **url** `string | URL | object`
+* **options** [`StreamOptions`]
+  * **dispatcher** `Dispatcher` - Default: [getGlobalDispatcher]
+  * **method** `String` - Default: `GET`
+* **factory** `Dispatcher.stream.factory`
+
+Returns a promise with the result of the `Dispatcher.stream` method.
+
+`url` may contain pathname. `options` may not contain path.
+
+Calls `options.dispatcher.stream(options, factory)`.
+
+See [Dispatcher.stream](docs/api/Dispatcher.md#dispatcherstream) for more details.
+
+## `undici.pipeline(url, options, handler): Duplex`
+
+Arguments:
+
+* **url** `string | URL | object`
+* **options** [`PipelineOptions`]
+  * **dispatcher** `Dispatcher` - Default: [getGlobalDispatcher]
+  * **method** `String` - Default: `GET`
+* **handler** `Dispatcher.pipeline.handler`
+
+Returns: `stream.Duplex`
+
+`url` may contain pathname. `options` may not contain path.
```

Comment:
Is `url` always normalized into a URL instance?

Reply:
`may` or `must`?

Reply:
May

Reply:
> Is url always normalized into a URL instance?

Awesome. What about calling `decodeURI(string)` in order to make sure things like `'https://example.com/%AA'` will throw? This might have a performance impact.

Reply:
Im not sure. @mcollina @dnlup wdyt?

Reply:
@szmarczak I think that's a separate issue though. Want to open a PR or issue for that?